src/loader.rs

```diff
@@ -0,0 +1,87 @@
+extern crate libloading;
+use self::libloading::{Library, Symbol};
+use super::driver::{Driver, DriverSchema};
+use std::collections::HashMap;
+
+pub type Result<T> = ::std::io::Result<T>;
+
+/// Trait implemented by anthing that wishes to load a driver at runtime.
+pub trait DriverLoader {
+  fn get_driver<'a>(&self) -> Option<&'a Driver>;
+  fn get_schema<'a>(&self) -> Option<&'a DriverSchema>;
+}
+
+pub struct Loader {
+  names: HashMap<String, String>,
+  libs: HashMap<String, Library>,
+  drivers: HashMap<String, Box<DriverLoader>>
+}
+
+impl Loader {
+  pub fn new() -> Self {
+    Loader {
+      names: HashMap::new(),
+      libs: HashMap::new(),
+      drivers: HashMap::new()
+    }
+  }
+
+  fn get_loc(&self, name: String) -> String {
+    self.names.get(&name).unwrap().clone()
+  }
+
+  fn load_lib(&mut self, loc: String) {
+    let lib = match Library::new(loc.clone()) {
+      Ok(lib) => lib,
+      Err(e)  => panic!("Failed to load lib '{}' due to:\n{}", loc, e)
```

Comment:
What do you think about returning `Result` and handling errors in a root function vs. random `panic!`s? (`try!` may help here)

I'm not sure which is the better pattern.